app/uk/gov/hmrc/apiplatform/modules/mfa/controllers/profile/MfaController.scala

```diff
@@ -99,43 +119,78 @@ class MfaController @Inject()(
         .fill(form)
         .withError(key = "accessCode", message = "You have entered an incorrect access code")
 
-      BadRequest(accessCodeView(mfaAccessCodeForm, mfaId))
+      BadRequest(authAppAccessCodeView(mfaAccessCodeForm, mfaId))
     }
 
     MfaAccessCodeForm.form.bindFromRequest.fold(
-      form => Future.successful(BadRequest(accessCodeView(form, mfaId))),
-      form => {
-        for {
+      form => Future.successful(BadRequest(authAppAccessCodeView(form, mfaId))),
+      form => for {
           mfaResponse <- mfaService.enableMfa(request.userId, mfaId, form.accessCode)
-          result = {
-            if (mfaResponse.totpVerified) logonAndComplete()
-            else invalidCode(form)
-          }
+          result = if (mfaResponse.totpVerified) logonAndComplete() else invalidCode(form)
         } yield result
-      })
+    )
   }
 
   def nameChangePage(mfaId: MfaId): Action[AnyContent] = atLeastPartLoggedInEnablingMfaAction { implicit request =>
     Future.successful(Ok(nameChangeView(MfaNameChangeForm.form, mfaId)))
   }
 
   def nameChangeAction(mfaId: MfaId): Action[AnyContent] = atLeastPartLoggedInEnablingMfaAction { implicit request =>
-
     MfaNameChangeForm.form.bindFromRequest.fold(
       form => Future.successful(BadRequest(nameChangeView(form, mfaId))),
       form => {
         thirdPartyDeveloperMfaConnector.changeName(request.userId, mfaId, form.name) map {
-          case true => Redirect(uk.gov.hmrc.apiplatform.modules.mfa.controllers.profile.routes.MfaController.authAppSetupCompletedPage())
-          case false => InternalServerError("Failed to change MFA name")
+          case true => Redirect(routes.MfaController.authAppSetupCompletedPage())
+          case false => internalServerErrorTemplate("Failed to change MFA name")
         }
       }
     )
   }
 
   def authAppSetupCompletedPage: Action[AnyContent] = atLeastPartLoggedInEnablingMfaAction { implicit request =>
-    Future.successful(Ok(authAppSetupCompletedView()))
+    thirdPartyDeveloperConnector.fetchDeveloper(request.userId).map {
```

Comment:
You have a UserRequest here so that should contain a Developer in the DeveloperSession.
Is that not enough?
Do you need to look up the developer again?

Reply:
The developer in the DeveloperSession didn't seem to have the updated mfaDetails that we'd saved. So the developer needs to be fetched again.

Reply:
That's a pain - I assume it's the developer from the session flow :(

Reply:
Yeah it's the same developer who logged in initially. But it has the old developer data.